### PR TITLE
Rework js files to leverage es6 imports

### PR DESCRIPTION
**Pull request by sgara**
Not a crucial PR on the road to webpack, but now that we use proper es6 imports, I figured we could clean things up a bit. Also renamed js files for consistency as some used hypens, others used underscores (I picked hypens). This also removes the ugly global ActiveAdmin variable :)

After this one, I think we can setup package distribution through npm. I can start it and give over the ownership afterward if you want.

- [x] setup npm packaging and rollup
- [ ] **rework js files to remove the use of ActiveAdmin global constant**
- [ ] **release the npm package including assets ready to be imported by webpack**
- [ ] add use_webpacker switch

### CODE REVIEW

File: app/assets/javascripts/active_admin/base.js [hunk 1]
```diff
@@ -345,184 +427,81 @@
...
+  function _inheritsLoose(subClass, superClass) {
+    subClass.prototype = Object.create(superClass.prototype);
+    subClass.prototype.constructor = subClass;
+    subClass.__proto__ = superClass;
+  }
+  var TableCheckboxToggler = function(_CheckboxToggler) {
+    _inheritsLoose(TableCheckboxToggler, _CheckboxToggler);
     function TableCheckboxToggler() {
-      return _ActiveAdmin$Checkbox.apply(this, arguments) || this;
+      return _CheckboxToggler.apply(this, arguments) || this;
```
Comment (deivid-rodriguez) [inline app/assets/javascripts/active_admin/base.js]: I think the rename of this parameter unrelated to the commit including it, so I would split the rename to a previous commit.

**Comment by deivid-rodriguez** [comment]
Regarding the `npm` package, it sounds good to me, whatever it's easier to you works :+1:.

Thanks for doing this work!

File: app/assets/javascripts/active_admin/base.js (hunk 1 above)
Comment (sgara) [reply app/assets/javascripts/active_admin/base.js]: This comes from changing `ActiveAdmin. CheckboxToggler = ...` to `export default CheckboxToggler;` which is in this same commit. That's on the compiled side, so no, for this one, it's one the right commit :)

Comment (deivid-rodriguez) [reply app/assets/javascripts/active_admin/base.js]: Right! :+1:

**Comment by sgara** [comment]
@deivid-rodriguez - @javierjulio also maybe? - do you guys already have an npm account? I'll need your username or email to invite you once I finish setting things up

**Comment by deivid-rodriguez** [comment]
I don't think I have an `npm` account. I guess I should register one so that you can invite me, right?

**Comment by sgara** [comment]
Yep, you can create one here: https://www.npmjs.com/signup

**Comment by deivid-rodriguez** [comment]
Done! I used the email present in activeadmin's gemspec!

**Comment by sgara** [comment]
I have minor changes to do on scss files (like using relative imports) before publishing the package. So we can merge this one and open yet another PR with remaining changes for publication

**Comment by javierjulio** [comment]
@sgara yes, I have an npm account. Can you please invite jjfutbol? Thanks!

**Comment by javierjulio** [comment]
Thanks! ❤️

**Pull request merged**